source/base/ByteBuffer.ooc

```diff
@@ -15,21 +16,24 @@ ByteBuffer: class {
 	_size: Int
 	_referenceCount: ReferenceCounter
 	_ownsMemory: Bool
+	_allocator: AbstractAllocator = null
 	pointer ::= this _pointer
 	size ::= this _size
 	referenceCount ::= this _referenceCount
 
-	init: func (=_pointer, =_size, ownsMemory := false) {
+	init: func (=_pointer, =_size, ownsMemory := false, allocator := null as AbstractAllocator) {
 		this _referenceCount = ReferenceCounter new(this)
 		this _ownsMemory = ownsMemory
+		this _allocator = allocator ?? MallocAllocator new() as AbstractAllocator
 	}
 	free: override func {
 		if (this _referenceCount != null)
 			this _referenceCount free()
 		this _referenceCount = null
 		if (this _ownsMemory)
-			memfree(this _pointer)
+			this _allocator deallocate(this _pointer)
 		this _pointer = null
+		this _allocator free()
```

Comment:
I think it's quite natural for the buffer to take ownership of the allocator. @emilwestergren can you think of a case were this would be undesirable?